datafusion/core/src/physical_plan/planner.rs

```diff
@@ -1739,8 +1739,6 @@ mod tests {
             col("c1").and(col("c1")),
             // u8 AND u8
             col("c3").and(col("c3")),
-            // utf8 = u32
```

Comment:
Why remove this case?

Reply:
utf8 and u32 now have coerced type (utf8).